src/app/server/java/AndroidAppServerWrapper.cpp

```diff
@@ -51,10 +51,7 @@ CHIP_ERROR ChipAndroidAppInit(void)
     err = chip::Server::GetInstance().Init(nullptr, CHIP_PORT, CHIP_UDC_PORT);
     SuccessOrExit(err);
 
-    if (!IsDeviceAttestationCredentialsProviderSet())
-    {
-        SetDeviceAttestationCredentialsProvider(Examples::GetExampleDACProvider());
-    }
+    ApplicationInit();
```

Comment:
Why does the app not provide a DACProvider to the startApp/init calls? Then this can call `SetDeviceAttestationCredentialsProvider` with the given pointer/value.

Reply:
Because the ApplicationInit is not only for init DAC, but also for many others like SetExtendedDiscoveryTimeoutSecs, please ref for examples/all-clusters-app/linux/main-common.cpp and examples/ota-requestor-app/linux/main.cpp